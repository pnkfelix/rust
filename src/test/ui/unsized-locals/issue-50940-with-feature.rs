#![feature(unsized_fn_params)]
//~^ WARN the feature `unsized_fn_params` is incomplete and may cause the compiler to crash

fn main() {
    struct A<X: ?Sized>(X);
    A as fn(str) -> A<str>;
    //~^ERROR the size for values of type `str` cannot be known at compilation time
}
