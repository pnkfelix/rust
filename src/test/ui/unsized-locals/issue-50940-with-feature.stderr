warning: the feature `unsized_fn_params` is incomplete and may cause the compiler to crash
  --> $DIR/issue-50940-with-feature.rs:1:12
   |
LL | #![feature(unsized_fn_params)]
   |            ^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default

error[E0277]: the size for values of type `str` cannot be known at compilation time
  --> $DIR/issue-50940-with-feature.rs:6:5
   |
LL |     A as fn(str) -> A<str>;
   |     ^ doesn't have a size known at compile-time
   |
   = help: within `main::A<str>`, the trait `std::marker::Sized` is not implemented for `str`
   = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
   = note: required because it appears within the type `main::A<str>`
   = note: the return type of a function must have a statically known size

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0277`.
